src/intrinsics/ecma262/Object.js

```diff
@@ -363,9 +363,11 @@ export default function(realm: Realm): NativeFunctionValue {
     // If we're in pure scope and the items are completely abstract,
     // then create an abstract temporal with an array kind
     if (realm.isInPureScope() && obj instanceof AbstractObjectValue) {
```

Comment:
It will be safer it this entire section of code moved to the front of this function.